src/sdk/pynni/nni/compression/torch/compressor.py

```diff
@@ -114,6 +115,48 @@ def new_forward(*inputs):
 
         layer.module.forward = new_forward
 
+    def export_model(self, model, save_path=None, onnx_path=None, input_shape=None):
+        """Pruned model can be exported with `export_model`.
+        `save_path` must be specified for saving model's `state_dict` and `mask_dict`.
+        `onnx_path` can be optionally specified for saving model to `.onnx` format.
+        `input_shape` is needed for exporting model to `.onnx` format.
+        """
+        export_dict = {}
+        export_dict['state_dict'] = model.state_dict()
+        mask_dict = {}
+        for name, value in model.state_dict():
+            found = False
+            for layer_name, mask in self.mask_list:
+                if name == 'module.' + layer_name + '.weight' or layer_name + '.weight':
+                    mask_dict[layer_name] = self.mask_list[layer_name]
+                    found = True
+                    break
+            if not found:
+                mask_dict[name] = None
+        export_dict['mask_dict'] = mask_dict
+        for name, m in model.named_modules():
+            mask = self.mask_list.get(name)
+            if mask is not None:
+                mask_sum = mask.sum().item()
+                mask_num = mask.numel()
+                _logger.info('Layer: {}  Sparsity: {}'.format(name, 1 - mask_sum / mask_num))
+                print('Layer: {}  Sparsity: {}'.format(name, 1 - mask_sum / mask_num))
+                m.weight.data = m.weight.data.mul(mask)
+            else:
+                _logger.info('Layer: {}  Sparsity: {}'.format(name, 0))
+                print('Layer: {}  Sparsity: {}'.format(name, 0))
+        assert save_path is not None, 'Save Path must be specified'
```

Comment:
suggest to separate mask and model to two files

Reply:
@QuanluZhang actually we often save checkpoints by saving optimizer state_dict and model state_dict to one file while using pytorch, here we can get the dicts like this
```
prune_dict = torch.load('prune_dict.pth')
state_dict = prune_dict['state_dict']
mask_dict = prune_dict['mask_dict']
```